pkg/deploy/automation.go

```diff
@@ -20,21 +20,16 @@ import (
 	"fmt"
 	"github.com/dynatrace/dynatrace-configuration-as-code/internal/idutils"
 	"github.com/dynatrace/dynatrace-configuration-as-code/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/template"
 	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/client/automation"
 	config "github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2"
 	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/parameter"
 )
 
-//go:generate mockgen -source=automation.go -destination=automation_mock.go -package=deploy automationClient
+//go:generate mockgen -source=automation.go -destination=automation_mock_client.go -package=deploy automationClient
```

Comment:
Is there any reason for renaming the destination file & adding a new gitignore entry?
Why can't this continue the convention of mocks ending in _mock.go?

Reply:
dummy client for automation is moved to `automation_dummy_client.go`. This was done to match the naming as they're in the same package. 

If it is necessary to keep the current convention that mocks generated files, have _moch.go suffix, and it'll be returned.

Reply:
It's not necessary, but personally, I some times remove all mocks as `**/*_mock.go` when switching between branches that have added/removed certain mocks. 

So them following the same convention is helpful

Reply:
I'll change that in a separate PR.
I tried to move automation in the deploy package to a separate package like in the download package. But it isn't so easy.

Reply:
I also just saw the change and was wondering why. As you stated, please move it back to _mock in a followup PR, or even this one. Should be a simple renaming, and we would not need to touch the .gitignore